pyvista/core/filters.py

```diff
@@ -1599,14 +1599,58 @@ def decimate_boundary(dataset, target_reduction=0.5):
         return dataset.extract_geometry().triangulate().decimate(target_reduction)
 
 
+    def sample_over_line(dataset, pointa, pointb, resolution=None, scalars=None):
+        """Sample a dataset along a high resolution line.
+
+        Parameters
+        ----------
+        pointa : np.ndarray or list
+            Location in [x, y, z].
+
+        pointb : np.ndarray or list
+            Location in [x, y, z].
+
+        resolution : int
+            number of pieces to divide line into. Defaults to number of cells
+            in the input mesh. Must be a positive integer.
+
+        scalars : str
+            The string name of the variable in the input dataset to probe. The
+            active scalar is used by default.
+
+        Return
+        ------
+        distance : np.ndarray
+            The distance along the line at the sampled points.
+        
+        values : np.ndarray
+            The values of scalars at the sampled points
+        """
+        if resolution is None:
+            resolution = int(dataset.n_cells)
+        if not isinstance(resolution, int) or resolution < 0:
+            raise RuntimeError('`resolution` must be a positive integer, not {}'.format(type(resolution)))
```

Comment:
I'd expect the resolution checking to happen in `pyvista.Line`.

Reply:
Ah good point! It would be good to go ahead and move that check over to thee `pyvista.Line` function

Reply:
I did nothing but refactor the code, so this was original, but it is always good to clean things up.  Let me dig deeper, I suppose this is a chance to learn more about pyvista internals.

Reply:
Yep, this one is on me when I originally wrote this filter. If you could move that check over to the following function, it would be a help!

https://github.com/pyvista/pyvista/blob/86a45088696f157c9c5751f214d4feb49a1091fa/pyvista/utilities/geometric_objects.py#L326

Reply:
With the current version if I do

```py
pv.Line([0, 0, 0], [0, 0, 1], resolution=10.5)
```

I get a helpful error already from presumably vtk
```txt
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/redacted/pyvista/pyvista/utilities/geometric_objects.py", line 348, in Line
    src.SetResolution(resolution)
TypeError: SetResolution argument 1: integer argument expected, got float
```

Do you still prefer a check inside pyvista with RunTimeError?

Reply:
I didn't touch this part yet as I'm not sure what you want.  My understanding is that `ValueError` is for arguments of the wrong type, which seems to fit here.

Reply:
ah that's a good point - then I would just let that chack from vtk be it

Reply:
i.e. no need to even have a check inside pyvista

Reply:
ok more subtleties....

```py
pv.Line([0, 0, 0], [0, 0, 1], resolution=0)
pv.Line([0, 0, 0], [0, 0, 1], resolution=-1)
```

All of these lines return a line with two points.  I think we should still check the magnitude to be positive in `Line`, which is in the newest commit I pushed.